backend/tessend/tessend.go

```diff
@@ -123,7 +149,41 @@ func (t *Tessend) Err() <-chan error {
 
 // Name returns the daemon name.
 func (t *Tessend) Name() string {
-	return "tessend"
+	return componentName
+}
+
+// Receiver returns the tessen receiver channel.
+func (t *Tessend) Receiver() chan<- interface{} {
+	return t.messageChan
+}
+
+func (t *Tessend) startMessageHandler() {
+	for {
+		msg, ok := <-t.messageChan
+		if !ok {
+			select {
+			case t.errChan <- errors.New("tessen message channel closed"):
+			default:
+			}
+			return
+		}
+
+		tessen, ok := msg.(*corev2.TessenConfig)
+		if !ok {
+			logger.WithField("msg", msg).Errorf("received non-TessenConfig on tessen message channel")
+			return
+		}
+
+		data := t.getDataPayload()
+		t.getTessenConfigMetrics(time.Now().UTC().Unix(), tessen, data)
+		logger.WithFields(logrus.Fields{
+			"url":                       t.url,
+			"id":                        data.Cluster.ID,
+			"opt-out":                   tessen.OptOut,
+			data.Metrics.Points[0].Name: data.Metrics.Points[0].Value,
+		}).Info("sending opt-out status event to tessen")
+		_ = t.send(data)
```

Comment:
Should this error be handled and logged?

Reply:
This function doesn't return an error, but errors are logged within in.